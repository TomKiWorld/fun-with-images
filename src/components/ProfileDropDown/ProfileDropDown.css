.profile-drop-down {
  position: relative;
}

.profile-drop-down .nav-drop-down {
  position: absolute;
  top: 3.5rem;
  right: 4rem;
  width: 11rem;
  font-size: 1rem;
  background-color: rgba(255, 255, 255, 0.5);
  text-align: left;
  border-radius: .25rem;
  box-shadow: 4px 4px 8px 0 rgba( 0, 0, 0, .2 );
}

.profile-drop-down .nav-drop-down ul {
  list-style: none;
  margin: 0;
  padding: 1rem;
}

.profile-drop-down .nav-drop-down ul li {
  padding: 0.5rem;
  border-radius: .25rem;
}

.profile-drop-down .nav-drop-down ul li:hover,
.profile-drop-down .nav-drop-down ul li:focus {
  background-color: #FFF;
}

.menu { 
  content: 'â‰¡';
  color: #CC0B78;
  font-size: 6rem;
  margin: 0 2rem;
}
