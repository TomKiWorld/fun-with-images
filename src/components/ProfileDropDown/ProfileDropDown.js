import React, { Component } from 'react';

import './ProfileDropDown.css';

class ProfileDropDown extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dropdownOpen: false
    };
  }

  toggle = (status) => {
    if (status) {
      this.setState({dropdownOpen: status})
    } else {
      this.setState(prevState => ({
        dropdownOpen: !prevState.dropdownOpen
      }));
    }
  }

  showDropDown = () => {
    const { onRouteChange  } = this.props;
    if (this.state.dropdownOpen) {
      return (
        <div 
          className='nav-drop-down'
          onMouseLeave={() => this.toggle(false)}
           >
            <ul>
              <li onClick={() => onRouteChange('home')} >Home</li>
              <li onClick={() => onRouteChange('profile')} >View Profile</li>
              <li onClick={() => onRouteChange('signin')} >Sign out</li>
            </ul>
        </div>
      )
    }
  }

  render() {
    return (      
      <div className='profile-drop-down tc'>
        <div aria-expanded={this.state.dropdownOpen}>
          <p
            className='menu'
            onClick={this.toggle}
            >â‰¡</p>
        </div>
        {this.showDropDown()}
      </div>
    )
  }
}

export default ProfileDropDown;
