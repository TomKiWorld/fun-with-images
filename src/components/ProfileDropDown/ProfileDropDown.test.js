import { shallow } from 'enzyme';
import React from 'react';
import ProfileDropDown from './ProfileDropDown';

describe('ProfileDropDown Component', () => {
  const wrapper = shallow(<ProfileDropDown />);

  it('Expect to render ProfileDropDown Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
