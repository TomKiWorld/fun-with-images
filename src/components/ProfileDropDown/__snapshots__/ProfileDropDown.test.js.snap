// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDropDown Component Expect to render ProfileDropDown Component 1`] = `ShallowWrapper {}`;
